events/api.py

```diff
@@ -2578,76 +2578,64 @@ def _filter_event_queryset(queryset, params, srs=None):  # noqa: C901
             local=True,
         )
 
-    val = params.get("local_ongoing_OR", None)
-    if val:
-        rc = _terms_to_regex(val, "OR")
-        ids = {
-            k
-            for k, v in cache.get("local_ids").items()
-            if rc.search(v, concurrent=True)
-        }
-        queryset = queryset.filter(id__in=ids)
+    def get_queryset_from_cache(params, param, cache_name, operator, queryset):
```

Comment:
Any reason for this and `get_queryset_from_cache_many` being located in the middle of the function? 🤔 Personally, I'd prefer them being in the beginning of the function. Even better if they're outside the function, since then you can write unit tests for them.

Reply:
Not other than that they're local functions. But moved them away from the parent function. Any tips how to use the cache in tests? Didn't find any examples using it.

Reply:
Of course, I'm not quite sure if changing the cache back-end affects other tests. I sure hope not, but if it does, then mocking might make more sense.

Reply:
Or maybe a fixture like this...
```python
from django.core.cache import cache

@pytest.fixture(autouse=True)
def teardown_cache():
    yield
    cache.clear()
```
Then you could just use Django's cache instead of requesting a fixture. Should be safer too. Assuming that caches aren't automatically cleared between tests already.

Reply:
Thanks for the good tips! Added tests for those new functions.